### PR TITLE
Issue #1905 - VRead returns error when invalid search parameters used

### PR DESCRIPTION
**Pull request by tbieste**
Updated vread to support the _elements and _summary search parameters, and return an error when invalid search parameters are used.

Signed-off-by: Troy Biesterfeld tbieste@us.ibm.com

### CODE REVIEW

File: fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java [hunk 1]
```diff
@@ -1312,6 +1312,35 @@ protected static boolean isAllowed(Type type, Modifier modifier) {
         return result;
     }
 
+    /**
+     * Parse query parameters for read and vread.
+     * @param resourceType the resource type
+     * @param queryParameters the query parameters
+     * @param queryString the query string
```
Comment (michaelwschroeder) [inline fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java:1319]: can remove this line

**Review by michaelwschroeder — APPROVED** [review]

**Comment by michaelwschroeder** [comment]
LGTM - just one minor comment

File: fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java (hunk 1 above)
Comment (tbieste) [reply fhir-search/src/main/java/com/ibm/fhir/search/util/SearchUtil.java]: Removed.

**Review by lmsurpre — APPROVED** [review]
LGTM

**Comment by tbieste** [comment]
Comments addressed. Ready to merge.

**Pull request merged**